FirebaseAuth/Tests/Sample/Sample/MainViewController+Passkey.m

```diff
@@ -0,0 +1,105 @@
+                                       action:^{ [weakSelf passkeySignUp]; }],
+    [StaticContentTableViewCell cellWithTitle:@"Sign In With Passkey"
+                                       action:^{ [weakSelf passkeySignIn]; }],
+    [StaticContentTableViewCell cellWithTitle:@"Enroll with Passkey"
+                                       action:^{ [weakSelf passkeyEnroll]; }],
+  ]];
+}
+
+- (void)passkeySignUp {
+  // sign in anoymously
+  [[AppManager auth] signInAnonymouslyWithCompletion:^(FIRAuthDataResult *_Nullable result,
+                                                       NSError *_Nullable error) {
+    if (error) {
+      [self logFailure:@"sign-in anonymously failed" error:error];
+    } else {
+      [self logSuccess:@"sign-in anonymously succeeded."];
+      [self log:[NSString stringWithFormat:@"User ID : %@", result.user.uid]];
+      [self passkeyEnroll];
+    }
+  }];
+  
+}
+
+- (void)passkeyEnroll {
+  FIRUser *user = FIRAuth.auth.currentUser;
+  if (!user) {
+    [self logFailure:@"Please sign in first." error:nil];
+    return;
+  }
+  [self showTextInputPromptWithMessage:@"passkey name"
+                          keyboardType:UIKeyboardTypeEmailAddress
+                       completionBlock:^(BOOL userPressedOK, NSString *_Nullable passkeyName) {
+    if (@available(iOS 16.0, macOS 12.0, tvOS 16.0, *)) {
+      [user startPasskeyEnrollmentWithName:passkeyName completion:^(ASAuthorizationPlatformPublicKeyCredentialRegistrationRequest * _Nullable request, NSError * _Nullable error) {
+        if (request) {
+          ASAuthorizationController *controller = [[ASAuthorizationController alloc] initWithAuthorizationRequests: [NSMutableArray arrayWithObject:request]];
+          controller.delegate = self;
+          controller.presentationContextProvider = self;
+          [controller performRequests];
+        } else if (error) {
+          [self logFailure:@"Passkey enrollment failed" error:error];
+        }
+      }];
+    } else {
+      [self log:@"OS version is not supported for this action."];
+    }
+  }];
+
+}
+
+- (void)passkeySignIn {
+  if (@available(iOS 16.0, macOS 12.0, tvOS 16.0, *)) {
+    [[AppManager auth] startPasskeySignInWithCompletion:^(ASAuthorizationPlatformPublicKeyCredentialAssertionRequest * _Nullable request, NSError * _Nullable error) {
+      if (request) {
+        ASAuthorizationController *controller = [[ASAuthorizationController alloc] initWithAuthorizationRequests: [NSMutableArray arrayWithObject:request]];
+        controller.delegate = self;
+        controller.presentationContextProvider = self;
+        [controller performRequestsWithOptions:ASAuthorizationControllerRequestOptionPreferImmediatelyAvailableCredentials];
+      }
+    }];
+  } else {
+    [self log:@"OS version is not supported for this action."];
+  }
+  
+}
+
+- (ASPresentationAnchor)presentationAnchorForAuthorizationController:(ASAuthorizationController *)controller API_AVAILABLE(ios(13.0)){
```

Comment:
Added this because "unrecognized selector sent to instance ..." exception.  However this raises a warning of "Category is implementing a method which will also be implemented by its primary class".  Any suggestions? @renkelvin

Reply:
You can try move this method to MainViewController.m